src/Avalonia.Base/Controls/ResourcesChangedToken.cs

```diff
@@ -0,0 +1,11 @@
+using System.Threading;
+
+namespace Avalonia.Controls;
+
+internal record struct ResourcesChangedToken(int SequenceNumber)
+{
+    private static int s_lastSequenceNumber;
```

Comment:
We are planning to have multiple dispatcher support at some point, please, mark as ThreadStatic or use interlocked increment apis